src/mapValues.test.ts

```diff
@@ -116,7 +116,7 @@ describe("typing", () => {
 
       mapValues(userValues, (value, key) => {
         expectTypeOf(value).toEqualTypeOf<number>();
-        expectTypeOf(key).toEqualTypeOf<`${UserID}`>();
+        expectTypeOf(key).toMatchTypeOf<`${UserID}`>();
```

Comment:
**question:** why doesn't `toEqualTypeOf` work here?

i think units should always assert the strongest possible assertion, and in this case i can't see how `key` can be narrower than `${UserID}`

Reply:
I got an error when using `toEqualTypeOf`, which I suspect has to do with type inference (?). I got it to work with `assertType` but it is not being used in codebase, so I changed it with `toMatchTypeOf`. Have you encountered a similar issue before ?

Reply:
strange indeed… i'm fine with this for now

Reply:
Let's add a TODO so we can test this out separately without delaying this PR